assets/js/theme/custom/add-reviews-to-cart-items.js

```diff
@@ -1,6 +1,22 @@
 /**
  * Add reviews to cart items via GraphQL
  */
+
+function setStarRating(rating) {
+    const stars = document.querySelectorAll('.c-star')
+    stars.forEach((star, index) => {
+        const fillValue = Math.max(0, Math.min(100, (rating - index) * 100))
+        const gradientId = `star-part-${index + 1}`
+        const gradient = star.querySelector(`#${gradientId}`)
+        if (gradient) {
+            const stops = gradient.querySelectorAll('stop')
+            stops.forEach((stop) => {
+                stop.setAttribute('offset', `${fillValue}%`)
```

Comment:
Even tho the code works, it's not how the `stop` should be set according to documentation. However you could simply a code and have only two `stops` element
```
  <stop offset="50%" class="js-fill c-star-fill"></stop>
  <stop offset="0" stop-color="white"></stop>
```

where the 1st one determines how much of a space needs to be filled, and the 2nd one is always set to 0; 

Also, take a look at `fillValue` sometimes it gives weird values e.g. for rating 4.6 the last star has filling set to `59.999999999999964%`

Reply:
Can you please review my updates again? I am not sure if I follow your comments regarding the `stop`.